pilot/pkg/networking/core/v1alpha3/cluster_test.go

```diff
@@ -788,13 +791,15 @@ func TestConditionallyConvertToIstioMtls(t *testing.T) {
 		Sni:               "custom.foo.com",
```

Comment:
This test is focused on testing checking the `tlsSettings` proto (istio proto). I would think we should have test demonstrating how the `tls_context` and `transport_socket_matches` been set correctly all together, since they together decide the final client behavior.

Or maybe it's not in this cluster_test.go? Somewhere else?